src/app/product-add/product-add.component.ts

```diff
@@ -12,38 +15,32 @@ export class ProductAddComponent implements OnInit {
     name : new FormControl('', Validators.required),
     category: new FormControl('', Validators.required),
     image: new FormControl('', Validators.required),
-    price: new FormControl('', [Validators.required, Validators.min(1), Validators.pattern("^[1-9]*$")]),
+    price: new FormControl('', [Validators.required, Validators.min(1), Validators.pattern("^[0-9]*$")]),
     description: new FormControl('', Validators.required)
   });
 
-  constructor(private productService: ProductsService) { }
+  constructor(private location: Location, private store: Store<AppState>) { }
 
   ngOnInit(): void {
   }
 
-  refreshForm() {
-    this.addForm.setValue({
-      name: "",
-      category: "",
-      image: "",
-      price: "",
-      description: ""
-    });
+  onSubmit() : void {
+    if ( this.addForm.value.name && this.addForm.value.price && this.addForm.value.category && this.addForm.value.image && this.addForm.value.description) {
+      const newProduct = {
+        id: Math.random(),
+        name: this.addForm.value.name,
+        category: this.addForm.value.category,
+        image: this.addForm.value.image,
+        price: parseInt(this.addForm.value.price),
+        description: this.addForm.value.description
+      }
+      this.store.dispatch(addProduct({product: newProduct}));
+      this.addForm.reset();
+      this.onLeave();
+    }
   }
 
-  onSubmit() {
-    let newProduct = {
-      name: this.addForm.value.name,
-      category: this.addForm.value.category,
-      image: this.addForm.value.image,
-      price: Number(this.addForm.value.price),
-      description: this.addForm.value.description
-    }
-    this.productService.addProduct(newProduct).subscribe(
-      (product) => { console.log(product);
-        alert("Product added successfully!");
-        this.refreshForm();},
-      (error) => {console.log(error);}
-    )
+  onLeave() : void{
+      this.location.back();
```

Comment:
I would move the redirect to the Effect. That way you have direct control over whether the result is successful or not, unlike here.